src/shared/lib/URLWrapper.ts

```diff
@@ -77,39 +75,28 @@ export default class URLWrapper<
         // even if they are not represented in browser url at the moment
         // they need to be there so that they will observable in the future upon assignment
         for (const property of this.properties) {
-            let value = (routing.location.query as QueryParamsType)[
+            let value: string | undefined | Object = (routing.location
+                .query as MapValues<QueryParamsType, string | undefined>)[
                 property.name
             ];
             if (_.isString(value)) {
                 // @ts-ignore
                 value = decodeURIComponent(value);
             }
-            initValues[property.name] = value;
+            if (property.nestedObjectProps) {
+                // nested object
+                value = value ? JSON.parse(value as string) : {};
+                for (const subprop of _.keys(property.nestedObjectProps!)) {
+                    value[subprop] = value[subprop] || undefined; // need to be there so that they will be observable in the future upon assignment
+                }
+            }
+            initValues[property.name] = value as any;
         }
 
         // consumers of wrapper read from this query
-        this.query = observable<QueryParamsType>(initValues as QueryParamsType);
-
-        //per reaction below, any update to URL will result in all properties being reset.
-        //to avoid update signal when properties haven't actually changed (set to existing value)
-        //we intercept changes and cancel them when old === new
-        intercept(this.query, change => {
```

Comment:
unless you somehow managed to accomplish this in another way, the intercept performs a crucial role.